pkg/route/validation/validation_test.go

```diff
@@ -1200,6 +1639,123 @@ func TestValidateRouteUpdate(t *testing.T) {
+		{description: "empty value", input: ``, validInput: false},
+		{description: "single character", input: `a`, validInput: true},
+		{description: "multiple characters", input: `abc`, validInput: true},
+		{description: "multiple words without escaped space", input: `abc def`, validInput: true},
+		{description: "multiple words with escaped space", input: `abc\ def`, validInput: true},
+		{description: "multiple words each word quoted", input: `"abc"\ "def"`, validInput: true},
+		{description: "multiple words each word quoted and with an embedded space", input: `"abc "\ "def "`, validInput: true},
+		{description: "multiple words each word one double quoted and other single quoted and with an embedded space", input: `"abc "\ 'def '`, validInput: true},
+		{description: "single % character", input: `%`, validInput: false},
+		{description: "escaped % character", input: `%%`, validInput: true}, // hmm?
+		{description: "escaped % and only a % character", input: `%%%`, validInput: false},
+		{description: "two literal % characters", input: `%%%%`, validInput: true},
+		{description: "zero percent", input: `%%%%%%0`, validInput: true},
+		{description: "escaped expression", input: `%%[XYZ.hdr(Host)]\ %[XYZ.hdr(Host)]`, validInput: true},
+		{description: "simple empty expression", input: `%[]`, validInput: false},
+		{description: "nested empty expressions", input: `%[%[]]`, validInput: false},
+		{description: "empty quoted value", input: `%{+Q}`, validInput: false},
+		{description: "quoted value", input: `%{+Q}foo`, validInput: false},
+		{description: "hdr with empty field", input: `%[XYZ.hdr()]`, validInput: false},
+		{description: "hdr with percent field", input: `%[XYZ.hdr(%)]`, validInput: false},
+		{description: "hdr with known field", input: `%[XYZ.hdr(Host)]`, validInput: true},
+		{description: "hdr with syntax error", input: `%[XYZ.hdr(Host]`, validInput: false},
+		{description: "hdr with url", input: `%[XYZ.hdr(Host)] http://url/hack`, validInput: true},
+
+		{description: "hdr with valid X-XSS-Protection value", input: `1;mode=block`, validInput: true},
+		{description: "hdr with valid Content-Type value", input: `text/plain,text/html`, validInput: true},
+		{description: "hdr with url", input: `text/plain,text/html`, validInput: true},
+
+		{description: "incomplete expression", input: `%[req`, validInput: false},
+		{description: "quoted field", input: `%[XYZ.hdr(%{+Q}Host)]`, validInput: false},
+		{description: "value with conditional expression", input: `%[XYZ.hdr(Host)] if foo`, validInput: true},
+		{description: "value with what looks like a conditional expression", input: `%[XYZ.hdr(Host)]\ if\ foo`, validInput: true},
+		{description: "unsuported fetcher", input: `%[date(3600),http_date]`, validInput: false},
+		{description: "nonexistent sample fetches or converters", input: `%[foo,lower]`, validInput: false},
+		{description: "nonexistent converters", input: `%[req.hdr(host),foo]`, validInput: false},
```

Comment:
It's not that the converter is nonexistent, it's because we only permit "lower" or "base64".